### PR TITLE
Fixed Leave Application Search Functionality

### PR DESCRIPTION
**Pull request by tejaszambre**
- Fixed show all users issue
- Fixed reset buttom
- Give access for all search filters of leave application to finance role

### CODE REVIEW

**Comment by tejaszambre** [comment]
[Leave Application Search Functionality Issue](https://trello.com/c/oNJFKNUN/60-leave-application-search-functionality-issue)

File: app/controllers/leave_applications_controller.rb [hunk 1]
```diff
@@ -142,13 +142,15 @@ def search_conditions
 
   def user_ids
     active_or_all_flag = params[:active_or_all_flag]
-    active_or_all_flag ||= "active" # show active users by default
-    if params[:project_id].present? and active_or_all_flag == "active"
-      UserProject.where(project_id: params[:project_id]).where(:end_date => nil).pluck(:user_id)
-    elsif params[:project_id].present? and active_or_all_flag == "all"
+    active_or_all_flag ||= 'active' # show active users by default
+    if params[:project_id].present? and active_or_all_flag == 'active'
+      UserProject.where(project_id: params[:project_id]).where(end_date: nil).pluck(:user_id)
```
Comment (Ashish944) [inline app/controllers/leave_applications_controller.rb:147]: Select only active team member (whose active status is true)

File: app/controllers/leave_applications_controller.rb [hunk 2]
```diff
@@ -142,13 +142,15 @@ def search_conditions
...
+    elsif params[:project_id].present? and active_or_all_flag == 'all'
       UserProject.where(project_id: params[:project_id]).pluck(:user_id)
     elsif params[:user_id].present?
-      [ params[:user_id] ]
+      [params[:user_id]]
+    elsif active_or_all_flag == 'all'
```
Comment (Ashish944) [inline app/controllers/leave_applications_controller.rb:152]: Move this condition before line 151 user_id condition

File: app/controllers/leave_applications_controller.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@ class LeaveApplicationsController < ApplicationController
   before_action :authenticate_user!
   load_and_authorize_resource except: [:create, :view_leave_status, :process_leave]
   before_action :authorization_for_admin, only: [:process_leave]
+  before_action :load_flag, only: [:view_leave_status, :user_ids]
```
Comment (Ashish944) [inline app/controllers/leave_applications_controller.rb:6]: user_ids is not a action method

File: app/controllers/leave_applications_controller.rb (hunk 1 above)
Comment (tejaszambre) [reply app/controllers/leave_applications_controller.rb:151]: Okay

**Review by Ashish944 — APPROVED** [review]

File: app/controllers/leave_applications_controller.rb (hunk 3 above)
Comment (tejaszambre) [reply app/controllers/leave_applications_controller.rb]: Okay

**Review by Ashish944 — APPROVED** [review]

**Review by swapnilchincholkar — APPROVED** [review]

**Pull request merged**